### DISCUSSION TITLE
Generating OpenAPI file for a specific directory

### DISCUSSION THREAD

**Discussion by JXenog** [smallrye/smallrye-open-api#1871] [Q&A] [answered]
Hey!

Currently, I'm researching a way to automatically generate OpenAPI Spec files from our java endpoints. However, I want an OpenAPI spec file that only contains the information about endpoints in a specific package, rather than the entire project. I was wondering if it was possible to configure this.

Thank you in advance!

**Comment by MikeEdgar** [comment, marked as answer]
@JXenog , there are several [MP OpenAPI configurations](https://download.eclipse.org/microprofile/microprofile-open-api-3.1.1/microprofile-openapi-spec-3.1.1.html#_core_configurations) that can be used to include or exclude packages and classes. You can try some combination to select just the endpoints you're interested in.

- `mp.openapi.scan.packages`
- `mp.openapi.scan.classes`
- `mp.openapi.scan.exclude.packages`
- `mp.openapi.scan.exclude.classes`
